examples/ucp_client_server.c

```diff
@@ -95,6 +96,95 @@ static struct {
  */
 static void usage(void);
 
+void buffer_free(ucp_dt_iov_t *iov)
+{
+    size_t idx;
+
+    for (idx = 0; idx < iov_cnt; idx++) {
+        if (iov[idx].buffer == NULL) {
+            continue;
+        }
+        mem_type_free(iov[idx].buffer);
+        iov[idx].buffer = NULL;
+        iov[idx].length = 0;
+    }
+    mem_type_free(iov);
+}
+
+ucp_dt_iov_t *buffer_malloc()
+{
+    size_t       idx;
+    ucp_dt_iov_t *iov;
+
+    iov = calloc(iov_cnt, sizeof(ucp_dt_iov_t));
+    CHKERR_ACTION(iov == NULL, "allocate memory\n", return NULL;);
+
+    for (idx = 0; idx < iov_cnt; idx++) {
+        iov[idx].length = test_string_length;
+        iov[idx].buffer = mem_type_malloc(iov[idx].length);
+        if (iov[idx].buffer == NULL) {
+            buffer_free(iov);
+            return NULL;
+        }
+        mem_type_memset(iov[idx].buffer, 0, iov[idx].length);
```

Comment:
why needed, if already filled in fill_buffer?

Reply:
@yosefe 
It keeps the same behaviour as the original code:
1. [clean buffer with zero](https://github.com/openucx/ucx/blob/v1.11.0-rc4/examples/ucp_client_server.c#L363)
2. [fill buffer with generate string](https://github.com/openucx/ucx/blob/v1.11.0-rc4/examples/ucp_client_server.c#L370)
I could remove it here if there's no need to keep the same behaviour as the original code.

Reply:
no need to keep previous behavior